scripts/entity/tactical/enemies/rf_bandit_bandit.nut

```diff
@@ -1,6 +1,8 @@
 this.rf_bandit_bandit <- ::inherit("scripts/entity/tactical/human", {
 	m = {
-		MyVariant = 0 // 1 for regular throwing weapons, 0 for not or throwing spear.
+		HasNet = 0, // 33% chance
+		IsRegularThrower = 0, // 50% chance
+		IsSpearThrower = 0 // 25% chance. only rolled if not regular thrower.
```

Comment:
I much prefer treating these variables as booleans instead for clarify of intent.
So have them be `false` by default.

And then change the roll-code to
this.m.IsSpearThrower = ::Math.rand(0, 3) == 0;

Reply:
Alternatively, if it is really your intent to capture the rolls, then you should rename the variables by having them include `Roll` at the start or end, instead of `has` and `is`

Reply:
I agree with Darxo. So the rolling could be like:
```
this.m.IsRegularThrower = ::Math.rand(1, 2) == 2; // this is a 50% chance.
```
So if the roll is 2 then `this.m.RegularThrower` will be `true` otherwise it'll be `false`. And so on with the other fields.

Reply:
This branch was tested in-game both before and after the boolean change.